lib/tupelo.js

```diff
@@ -289,6 +287,36 @@ class TupeloClient {
 
     return this.resolve(chainId, resolvePath)
   }
+  
+  /**
+   * Resolve data from a tip of a chain tree given a certain path.
+   *
+   * @param {string} chainId - The ID of the chain tree to retrieve the data from.
+   * @param {string} path - '/' delimited path into the chain tree where the data is stored
+   * @param {string} tip - The tip in question.
+   *
+   * @return {Promise<ResolveResponse, RpcError>}
+   */
+  async resolveAt(chainId, path, tip) {
```

Comment:
Overall looking good!

I did read this line and went back and forth whether I expected `tip` to be the first or last argument. I can see it going either way - what were your thoughts behind it being last?

Reply:
@brandonwestcott Actually there's no deep reasoning behind, just that I moved `tip` to the end of the argument list in Tupelo since it's an optional pointer, so I opted for symmetry here.